src/framework/diagnostics/glproblemsdetector.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "glproblemsdetector.h"
+
+#include <QFile>
+
+#include "global/internal/globalconfiguration.h"
+#include "global/logger.h"
+
+#include "log.h"
+
+using namespace muse::diagnostics;
+using namespace muse::logger;
+
+static const std::string BAD_MESSAGE = "Failed to build graphics pipeline state";
+
+//! NOTE This is message just for test
+//static const std::string BAD_PATTERN = "QML QQuickItem: Binding loop detected for property ";
+
+namespace muse::diagnostics {
+class GlLogDest : public LogDest
+{
+public:
+
+    GlLogDest()
+        : LogDest(LogLayout("")) {}
+
+    std::string name() const { return "GlProblemsDetector"; }
+    void write(const LogMsg& logMsg)
+    {
+        if (!enabled || isProblemDetected) {
+            return;
+        }
+
+        if (logMsg.message.find(BAD_MESSAGE) != std::string::npos) {
+            isProblemDetected = true;
+            std::cout << "found message: " << BAD_MESSAGE << std::endl;
+        }
+    }
+
+    bool enabled = false;
+    bool isProblemDetected = false;
+};
+}
+
+GlProblemsDetector::GlProblemsDetector()
+{
+    m_logDest = new GlLogDest();
+    m_logDest->enabled = true;
+
+    Logger* logger = Logger::instance();
+    logger->addDest(m_logDest);
+}
+
+GlProblemsDetector::~GlProblemsDetector()
+{
+    //! NOTE log dest will deleted by logger
+    m_logDest->enabled = false;
```

Comment:
Can we maybe add `Logger::removeDest` and call that here? In order to have completely no influence on performance once the program is running

Reply:
yes, that would be better...
I thought about it, but didn't dare...
I'll do it.